Duplicate bundled template Erlang Custom Template Variables.erl

```
Assertion failed: Duplicate bundled template Erlang Custom Template Variables.erl [jar:file:/Users/Aniruddha/Library/Application%20Support/IdeaIC2017.1/intellij-elixir/lib/erlang.jar!/fileTemplates/includes/Erlang Custom Template Variables.erl.ft, file:/Users/Aniruddha/Library/Application%20Support/IdeaIC2017.1/Erlang/classes/fileTemplates/includes/Erlang Custom Template Variables.erl.ft]

java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:168)
	at com.intellij.ide.fileTemplates.impl.FTManager.createAndStoreBundledTemplate(FTManager.java:204)
	at com.intellij.ide.fileTemplates.impl.FTManager.addDefaultTemplate(FTManager.java:195)
	at com.intellij.ide.fileTemplates.impl.FileTemplatesLoader.loadDefaultsFromRoot(FileTemplatesLoader.java:192)
	at com.intellij.ide.fileTemplates.impl.FileTemplatesLoader.loadDefaultTemplates(FileTemplatesLoader.java:152)
	at com.intellij.ide.fileTemplates.impl.FileTemplatesLoader.<init>(FileTemplatesLoader.java:94)
	at com.intellij.ide.fileTemplates.impl.FileTemplateSettings.<init>(FileTemplateSettings.java:48)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:103)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:80)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:63)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:228)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:239)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:206)
	at org.picocontainer.defaults.BasicComponentParameter.resolveInstance(BasicComponentParameter.java:77)
	at org.picocontainer.defaults.ComponentParameter.resolveInstance(ComponentParameter.java:114)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getConstructorArguments(CachingConstructorInjectionComponentAdapter.java:128)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:100)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:80)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:63)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:228)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:239)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:206)
	at com.intellij.openapi.components.ServiceManager.doGetService(ServiceManager.java:48)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:42)
	at com.intellij.ide.fileTemplates.FileTemplateManager.getInstance(FileTemplateManager.java:55)
	at com.intellij.codeInspection.defaultFileTemplateUsage.FileHeaderChecker.checkFileHeader(FileHeaderChecker.java:47)
	at com.intellij.codeInspection.defaultFileTemplateUsage.DefaultFileTemplateUsageInspection.checkFile(DefaultFileTemplateUsageInspection.java:64)
	at com.intellij.codeInspection.AbstractBaseJavaLocalInspectionTool$1.visitFile(AbstractBaseJavaLocalInspectionTool.java:88)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:229)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:81)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:69)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.runToolOnElements(LocalInspectionsPass.java:282)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitPriorityElementsAndInit$3(LocalInspectionsPass.java:250)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:133)
	at com.intellij.concurrency.ApplierCompleter.tryToExecAllList(ApplierCompleter.java:222)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:151)
	at com.intellij.concurrency.ApplierCompleter.tryToExecAllList(ApplierCompleter.java:222)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:151)
	at com.intellij.concurrency.ApplierCompleter.tryToExecAllList(ApplierCompleter.java:222)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:151)
	at com.intellij.concurrency.ApplierCompleter.tryToExecAllList(ApplierCompleter.java:222)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:151)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1147)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$1(ApplierCompleter.java:105)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:116)
	at com.intellij.concurrency.ApplierCompleter.lambda$compute$0(ApplierCompleter.java:96)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:242)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:96)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.pollAndExecCC(ForkJoinPool.java:1190)
	at java.util.concurrent.ForkJoinPool.helpComplete(ForkJoinPool.java:1879)
	at java.util.concurrent.ForkJoinPool.awaitJoin(ForkJoinPool.java:2045)
	at java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:390)
	at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:719)
	at java.util.concurrent.ForkJoinPool.invoke(ForkJoinPool.java:2616)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:63)
	at com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:57)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.visitPriorityElementsAndInit(LocalInspectionsPass.java:253)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.inspect(LocalInspectionsPass.java:223)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.collectInformationWithProgress(LocalInspectionsPass.java:132)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:83)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:438)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1153)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:431)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:430)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:406)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:242)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:404)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:165)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

Found https://github.com/JetBrains/gradle-intellij-plugin/issues/201, updating `org.jetbrains.intellij` to version "0.2.15" fixed the problem, now I need to get around 

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'Elixir'.
> Plugin org.jetbrains.erlang:0.9.958 is not compatible to IC-171.4694.70
```

Using the Erlang.jar directly must have hidden a lot of undeclared dependencies in my build.  Going down to version 0.9.939 made the build get farther, but not my `jps` namespace can't resolve any of the `org.jetbrains.annotations` like `org.jetbrains.annotations.NotNull`:

```
/Users/limhoff/git/KronicDeth/intellij-elixir/jps-shared/src/org/elixir_lang/jps/mix/JpsMixConfigurationExtension.java:3: error: cannot find symbol
import org.jetbrains.annotations.NotNull;
                                ^
  symbol:   class NotNull
  location: package org.jetbrains.annotations
```

--- a/build.gradle
+++ b/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-  id 'org.jetbrains.intellij' version "0.1.10"
+  id 'org.jetbrains.intellij' version "0.2.15"
   id 'de.undercouch.download' version "3.1.2"
 }
 
@@ -27,16 +27,30 @@ allprojects {
 
   apply plugin: 'org.jetbrains.intellij'
   intellij {
+    if (ideaVersion.startsWith("2017.1.")) {
+      plugins = ['org.jetbrains.erlang:0.9.939']
+    } else if (ideaVersion.startsWith("2016.3.")) {
+      plugins = ['org.jetbrains.erlang:0.9.912']
+    } else if (ideaVersion.startsWith("2016.2.")) {
+      plugins = ['org.jetbrains.erlang:0.8.913']
+    } else if (ideaVersion.startsWith("2016.1.")) {
+      plugins = ['org.jetbrains.erlang:0.7.876']
+    } else if (ideaVersion.startsWith("15.0.")) {
+      plugins = ['org.jetbrains.erlang:0.6.873']
+    } else {
+      throw new IllegalArgumentException("ideaVersion is not a recognized MAJOR.MINOR for IntelliJ Erlang compatibility")
+    }
+
     pluginName 'intellij-elixir'
     version ideaVersion
     downloadSources Boolean.valueOf(sources)
     updateSinceUntilBuild = false
+  }
 
-    publish {
-      username publishUsername
-      password publishPassword
-      channel publishChannel
-    }
+  publishPlugin {
+    username publishUsername
+    password publishPassword
+    channels publishChannels
   }
 
   def compilationPackages = ['org/intellij/elixir/build/**', 'org/intellij/elixir/jps/**']
@@ -64,10 +78,9 @@ allprojects {
 }
 
 repositories {
-  jcenter()
-  flatDir {
-    dirs 'libs'
-  }
+  maven { url 'http://dl.bintray.com/jetbrains/intellij-plugin-service' }
+  maven { url 'https://maven-central.storage.googleapis.com' }
+  mavenCentral()
 }
 
 dependencies {
@@ -75,7 +88,6 @@ dependencies {
   compile project('jps-shared')
   compile group: 'org.erlang.otp', name: 'jinterface', version: '1.6.1'
   compile group: 'commons-io', name: 'commons-io', version: '2.5'
-  compile name: 'erlang'
 
   testCompile group: 'org.mockito', name: 'mockito-core', version: '2.2.9'
   testCompile group: 'org.objenesis', name: 'objenesis', version: '2.4'
@@ -84,16 +96,6 @@ dependencies {
 project(':jps-builder') {
   dependencies {
     compile project(':jps-shared')
-  }
-
-  intellij {
-    type 'JPS'
-  }
-}
-
-project(':jps-shared') {
-  intellij {
-    type 'JPS'
   }
 }
 
@@ -105,27 +107,6 @@ idea {
   }
   module {
     generatedSourceDirs += file('gen')
-  }
-}
-
-task downloadIntellijErlang {
-  doLast {
-    download {
-      src "https://github.com/ignatov/intellij-erlang/releases/download/%23${intellijErlangRelease}/Erlang.${intellijErlangRelease}.zip"
-      dest "${rootDir}/cache/Erlang.${intellijErlangRelease}.zip"
-      overwrite false
-    }
-
-    copy {
-      from zipTree("${rootDir}/cache/Erlang.${intellijErlangRelease}.zip")
-      into "${rootDir}/cache/"
-      include 'Erlang/lib/erlang.jar'
-    }
-
-    copy {
-      from "${rootDir}/cache/Erlang/lib/erlang.jar"
-      into 'libs/'
-    }
   }
 }
 
@@ -187,7 +168,6 @@ task getQuoter {
 }
 
 compileJava {
-  dependsOn downloadIntellijErlang
 }
 
 compileTestJava {

--- a/gradle.properties
+++ b/gradle.properties
@@ -12,4 +12,4 @@ elixirVersion = 1.4.5
 quoterVersion = 0.1.1
 publishUsername =
 publishPassword =
-publishChannel =
+publishChannels = [default]

